File: app/src/main/java/com/mhmd/dribbblenotepad_1/data/model/Note.java
package com.mhmd.dribbblenotepad_1.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "note_table")
public class Note {

    @PrimaryKey(autoGenerate = true)
    int id;
    String title;
    String body;
    String date;
    int color;

    public Note(String title, String body, String date, int color) {
        this.title = title;
        this.body = body;
        this.date = date;
        this.color = color;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public String getBody() {
        return body;
    }

    public String getDate() {
        return date;
    }

    public int getColor() {
        return color;
    }
}

File: app/src/main/java/com/mhmd/dribbblenotepad_1/data/model/Note.kt
package com.mhmd.dribbblenotepad_1.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "note_table")
data class Note(
    @PrimaryKey(autoGenerate = true) val id: Int,
    val title: String = "",
    val body: String = "",
    val date: String = "",
    val color: Int = 0
)